qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -40,15 +37,79 @@ def _run_analysis(self, experiment_data: CompositeExperimentData, **options):
             QiskitError: if analysis is attempted on non-composite
                          experiment data.
         """
-        if not isinstance(experiment_data, CompositeExperimentData):
-            raise QiskitError("CompositeAnalysis must be run on CompositeExperimentData.")
+        composite_exp = experiment_data.experiment
+        component_exps = composite_exp.component_experiment()
+        if "component_job_metadata" in experiment_data.metadata:
+            component_metadata = experiment_data.metadata["component_job_metadata"][-1]
+        else:
+            component_metadata = [{}] * composite_exp.num_experiments
+
+        # Initialize component data for updating and get the experiment IDs for
+        # the component child experiments
+        component_ids = self._initialize_components(composite_exp, experiment_data)
+
+        # Compute marginalize data
+        marginalized_data = self._marginalize_data(experiment_data.data())
+
+        # Construct component experiment data
+        for i, (sub_data, sub_exp) in enumerate(zip(marginalized_data, component_exps)):
+            sub_exp_data = experiment_data.child_data(component_ids[i])
+
+            # Clear any previously stored data and add marginalized data
+            sub_exp_data._data.clear()
+            sub_exp_data.add_data(sub_data)
 
-        comp_exp = experiment_data.experiment
+            # Add component job metadata
+            sub_exp_data._metadata["job_metadata"] = [component_metadata[i]]
```

Comment:
This is intended to override the data. Since these containers are dynamically created and updated by analysis there is no reason to keep any previous job data (There is actually no need to have this as a list in base experiment since #463 , since you can't re-run in an existing container).